### DISCUSSION TITLE
how to use AP_STADISCONNECTED to reset ESP

### DISCUSSION THREAD

**Discussion by fikririzal** [espressif/arduino-esp32#7434] [Q&A] [answered]
ESP32

after I get this 2 massage:
22:52:38.302 -> [212495][V][WiFiGeneric.cpp:412] _arduino_event_cb(): AP Station Disconnected: MAC: 00:45:e2:9c:a3:d1, AID: 1
22:52:38.349 -> [212495][D][WiFiGeneric.cpp:929] _eventCallback(): Arduino Event: 13 - AP_STADISCONNECTED
my program was stuck and I have to press the reset button to Run normally, how to use that event to trigger reset ESP?

thankyou

**Comment by mrengineer7777** [comment, marked as answer]
You can register a WiFi event callback:
```
//-------------------Local function definitions-------------------
void WiFiEvent(WiFiEvent_t event, WiFiEventInfo_t info);
//-------------------Local function definitions-------------------

void setup() {
   wifi_event_id_t eventhandler_id = WiFi.onEvent(WiFiEvent);
}

void WiFiEvent(WiFiEvent_t event, WiFiEventInfo_t info) {
   if(event==ARDUINO_EVENT_WIFI_STA_DISCONNECTED) {
       //Do something based on reason in info.wifi_sta_disconnected.reason
   }
}
```

Note AP_STADISCONNECTED is actually a client disconnecting from your SoftAP.
